File: time-conversion/time-conversion.clj
(defn format-hh [unformatted-hh, ss-with-period]
  (let [is-pm? (str/includes? ss-with-period "PM")
        period-conversion (if is-pm? (+ 12 (read-string unformatted-hh)) unformatted-hh)]
    (case period-conversion
      24 "12"
      "12" "00"
      period-conversion)))

(defn timeConversion [s]
  (let [[unformatted-hh mm ss-with-period] (str/split s #"\:")
        hh (format-hh unformatted-hh ss-with-period)
        ss (subs ss-with-period 0 2)]
    (str/join [hh ":" mm ":" ss])))

File: time-conversion/time-conversion.js
// https://www.hackerrank.com/challenges/time-conversion/problem

const timeConversion = time => {
  const [unformattedHH, mm, ssWithPeriod] = time.split(':');

  const hh = (() => {
    const isPm = ssWithPeriod.includes('PM');

    const periodConversion = isPm ? 12 + +unformattedHH : unformattedHH;

    switch (periodConversion) {
      case 24:
        return '12';
      case '12':
        return '00';
      default:
        return periodConversion;
    }
  })();

  const ss = ssWithPeriod.slice(0, 2);

  return `${hh}:${mm}:${ss}`;
};
